cmd/otel-allocator/config/flags.go

```diff
@@ -69,8 +69,13 @@ func getListenAddr(flagSet *pflag.FlagSet) (string, error) {
 	return flagSet.GetString(listenAddrFlagName)
 }
 
-func getPrometheusCREnabled(flagSet *pflag.FlagSet) (bool, error) {
-	return flagSet.GetBool(prometheusCREnabledFlagName)
+func getPrometheusCREnabled(flagSet *pflag.FlagSet) (value bool, changed bool, err error) {
+	if changed = flagSet.Changed(prometheusCREnabledFlagName); !changed {
+		value, err = false, nil
```

Comment:
shouldn't the middle value just be false here?

Reply:
Yes, I could just do a plain `return` here, but imo this is more explicit. If you disagree, I can remove it, I don't feel too strongly about it.